### PR TITLE
[DomCrawler] Exposed getter for baseHref

### PR DESCRIPTION
**Pull request by a-ast**
| Q | A |
| --- | --- |
| Branch | master |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #15584 |
| License | MIT |
| Doc PR | n/a |

I know  that PR for this issue already created, but it is Hack Day, so I'm making my first contribution anyway.

### CODE REVIEW

**Comment by javiereguiluz** [comment]
@AAstakhov in other circumstances it would be weird to open a PR for some issue which already has a PR ... but you are right and we are in the Hack Day so you can open it. Maybe it gets closed without merging it ... but maybe the other one is closed. Let's wait and see. Thanks!

File: src/Symfony/Component/DomCrawler/Tests/CrawlerTest.php [hunk 1]
```diff
@@ -27,6 +27,12 @@ public function testConstructor()
         $this->assertCount(1, $crawler, '__construct() takes a node as a first argument');
     }
 
+    public function testGetBaseHref()
+    {
+        $crawler = new Crawler(null,  null, 'http://symfony.com');
+        $this->assertEquals('http://symfony.com', $crawler->getBaseHref());
```
Comment (patrick-mcdougle) [inline src/Symfony/Component/DomCrawler/Tests/CrawlerTest.php]: You could put the string `'http://symfony.com'` into a single variable, so one of them can't possibly be typo'd.

**Comment by AAstakhov** [comment]
@patrick-mcdougle, fixed CrawlerTest regarding to your suggestion.

**Comment by stof** [comment]
:+1: 

Status: reviewed

**Comment by nicolas-grekas** [comment]
:+1:
Status: reviewed
Note for an other reader: `$this->baseHref` is stateful in that in can change when `addHtmlContent` is called. This maybe the reason why the property has not getter. WDOT?

**Comment by fabpot** [comment]
@nicolas-grekas That's true in theory, but in practice, if you can `addHtmlContent()` several times, you end up with a broken crawler, so I don't think it matters too much.

**Comment by fabpot** [comment]
Thank you @AAstakhov.

**Pull request closed without merging**